metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage.py

```diff
@@ -299,58 +363,71 @@ def _get_upstream_lineage_info(
-            if not self.config.ignore_start_time_lineage
-            else 0,
-            end_time_millis=int(self.config.end_time.timestamp() * 1000),
-        )
 
         num_edges: int = 0
         self._external_lineage_map = defaultdict(set)
-        try:
-            for db_row in self.query(conn, query):
-                # key is the down-stream table name
-                key: str = self.get_dataset_identifier_from_qualified_name(
-                    db_row["DOWNSTREAM_TABLE_NAME"]
-                )
-                if not self._is_dataset_pattern_allowed(key, "table"):
-                    continue
-                self._external_lineage_map[key] |= {
-                    *json.loads(db_row["UPSTREAM_LOCATIONS"])
-                }
-                logger.debug(
-                    f"ExternalLineage[Table(Down)={key}]:External(Up)={self._external_lineage_map[key]} via access_history"
-                )
-        except Exception as e:
-            self.warn(
-                "external_lineage",
-                f"Populating table external lineage from Snowflake failed."
-                f"Please check your premissions. Continuing...\nError was {e}.",
+        if self.report.edition == SnowflakeEdition.STANDARD:
+            logger.info(
+                "Snowflake Account is Standard Edition. External Lineage Feature via Access History is not supported."
             )
+        else:
+            # Handles the case where a table is populated from an external location via copy.
+            # Eg: copy into category_english from 's3://acryl-snow-demo-olist/olist_raw_data/category_english'credentials=(aws_key_id='...' aws_secret_key='...')  pattern='.*.csv';
+            query: str = SnowflakeQuery.external_table_lineage_history(
+                start_time_millis=int(self.config.start_time.timestamp() * 1000)
+                if not self.config.ignore_start_time_lineage
+                else 0,
+                end_time_millis=int(self.config.end_time.timestamp() * 1000),
+            )
+
+            try:
+                for db_row in self.query(conn, query):
+                    # key is the down-stream table name
+                    key: str = self.get_dataset_identifier_from_qualified_name(
+                        db_row["DOWNSTREAM_TABLE_NAME"]
+                    )
+                    if not self._is_dataset_pattern_allowed(key, "table"):
+                        continue
+                    self._external_lineage_map[key] |= {
+                        *json.loads(db_row["UPSTREAM_LOCATIONS"])
+                    }
+                    logger.debug(
+                        f"ExternalLineage[Table(Down)={key}]:External(Up)={self._external_lineage_map[key]} via access_history"
+                    )
+            except SnowflakePermissionError:
+                error_msg = "Failed to get external lineage. Please grant permissions for SNOWFLAKE database. "
```

Comment:
What does "grant permissions for SNOWFLAKE database" mean? Let's say to grant permissions for accessing usage history and lineage right? I mean this should align with what we say in the tutorial

Reply:
"grant imported priviledges for SNOWFLAKE database" would be the exact message in snowflake terminology (relatable to snowflake user.

" grant permissions for accessing usage history and lineage" would be message from datahub's point of view and may not be super clear, I think.

Reply:
Generally( the other error messages too), the error message content is in terms of exact snowflake permission missing, rather than referring to a generic section in snowflake guide/docs.